src/api/03-user-experience/09-google-cast.md

```diff
@@ -0,0 +1,247 @@
+    // 5. When the `session` state change event is `SESSION_STARTED`,
+    //    we try to load the video.
+    if (state === 'SESSION_STARTED') {
+      const session = context.getCurrentSession();
+      const mediaInfo = new chrome.cast.media.MediaInfo(
+        'http://localhost/video.mp4',                // Provide a URL.
+        'video/mp4; codecs="avc1.640032,mp4a.40.2"', // MIME + codecs.
+      );
+      const request = new chrome.cast.media.LoadRequest(mediaInfo);
+
+      try {
+        // 6. Instruct the receiver application to load the media,
+        //    if it fails, be prepared to handle the error.
+        await session.loadMedia(request);
+      } catch (error) { /* Handle the error. */ }
+    }
+  }
+}
+</script>
+<script src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1"></script>
+
+<video src="video.mp4"></video>
+
+<!-- Custom element defined by the framework. -->
+<google-cast-launcher></google-cast-launcher>
+```
+
+## Metadata
+
+Receiver applications may expose additional information about the media in their UI, or provide additional features depending on metadata passed to them.
+
+The [Cast Application Framework] defines several [Metadata classes] that developers can use to convey information like the content title, thumbnail, release dates, authors etc.
+
+```js
+const mediaInfo = new chrome.cast.media.MediaInfo(
+  'http://localhost/video.mp4',                // Provide a URL.
+  'video/mp4; codecs="avc1.640032,mp4a.40.2"', // MIME + codecs.
+);
+
+// Image URLs need to be "wrapped" by the Image class.
+const thumbnail = new chrome.cast.Image('http://localhost/thumb.jpg');
+
+// Our video is a short clip, but if it was a full-length movie,
+// we would use `MovieMediaMetadata` here.
+mediaInfo.metadata = new chrome.cast.media.GenericMediaMetadata();
+mediaInfo.metadata.title = 'My Dog Chasing Geese';
+mediaInfo.metadata.images = [ thumbnail ];
+```
+
+## Captions and subtitles
+
+Media can contain additional [tracks of different types]. You can use text tracks to provide captions or subitles for your videos.
+
+```js
+const mediaInfo = new chrome.cast.media.MediaInfo(
+  'http://localhost/video.mp4',                // Provide a URL.
+  'video/mp4; codecs="avc1.640032,mp4a.40.2"', // MIME + codecs.
+);
+
+// Instantiate the Track object first.
+const captionsTrack = new chrome.cast.media.Track(
+  'captions-en', // Unique track identifier.
+  chrome.cast.media.TrackType.TEXT,
+);
+
+captionsTrack.trackContentId = 'http://localhost/captions-en.vtt';
+captionsTrack.subtype = chrome.cast.media.TextTrackType.CAPTIONS;
+captionsTrack.name = 'English CC';
+captionsTrack.language = 'en-US'; // RFC 5646 Language Tag.
+captionsTrack.trackContentType = 'text/vtt';
+
+// The `MediaInfo` object may contain several tracks of various types,
+// how and whether additional tracks are exposed in the UI depends
+// on the receiver application implementation.
+mediaInfo.tracks = [ captionsTrack ];
+```
```

Comment:
It would be good to remind people `mediaInfo` is used in `chrome.cast.media.LoadRequest(mediaInfo);`

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506.